### PR TITLE
Fix error span tag

### LINKED ISSUE #40: Errors not properly send to zipkin
Errors not properly send to zipkin due to a bad request http call.
The error tag must be set to a string with value "true" not the HTTP status code.

### PR DESCRIPTION
**Pull request by Landerstraeten**
Fixes #40

### CODE REVIEW

**Comment by jcchavezs** [comment]
I am not entirely sure this is an issue. Other server frameworks uses the status code for error:

- https://github.com/openzipkin/brave/blob/master/instrumentation/http/src/main/java/brave/http/HttpParser.java#L145
- https://github.com/openzipkin/zipkin-go/blob/master/middleware/http/server.go#L138

File: src/ZipkinBundle/Middleware.php [hunk 1]
```diff
@@ -122,7 +122,7 @@ public function onKernelTerminate(PostResponseEvent $event)
 
         $statusCode = $event->getResponse()->getStatusCode();
         if ($statusCode > 399) {
-            $span->tag(Tags\ERROR, $statusCode);
+            $span->tag(Tags\ERROR, 'true');
```
Comment (UrGuardian4ngel) [inline src/ZipkinBundle/Middleware.php]: Related to this change, I guess that there is a conflict between this assignment here, and the one in `ZipkinBundle\Middleware::onKernelException()`, which gets executed before this one...

https://github.com/jcchavezs/zipkin-instrumentation-symfony/blob/989ad9dc858ee02426954b96c651e5912dea81a1/src/ZipkinBundle/Middleware.php#L96-L98

Both assignments use the same key in the span's bagage items.

According to the [docs](https://opentracing.io/docs/overview/tags-logs-baggage/):

> Examples may include tag keys like `db.instance` to identify a database host, `http.status_code` to represent the HTTP response code, **or `error` which can be set to True** if the operation represented by the Span fails.

So maybe we should pick another key for the exception message, assigned to `Tags\Error` in #L96-L98 ?

Comment (jcchavezs) [reply src/ZipkinBundle/Middleware.php]: Hi @UrGuardian4ngel I think we are mixing two things here: The docs you are pointing are for `OpenTracing` while this is the zipkin instrumentation for symfony. Either way you are right there is some sort of unconsistency between the exception message and the error tag (status code). @adriancole do you have any input on this, what do other frameworks do?

Comment (adriancole) [reply src/ZipkinBundle/Middleware.php]: we use the error tag to include a short message, ideally human readable as opposed to an error code, but error code in worst case. For example, "Socket disconnected" "SocketException" "503" are all valid. Try not to embed huge things like stack traces, or things that clarify nothing like "true"

**Comment by jcchavezs** [comment]
In our case when there is an exception we include the exception message and
when there is a request with status code >399 we include the code, I know
symfony has a translation table from status code into message (see
https://github.com/symfony/http-foundation/blob/master/Response.php#L131)
but I am not sure that is more useful than the status code.

*As users, will that be more useful to you, people?*


Den fre. 22. mar. 2019, 01:05 skrev Adrian Cole <notifications@github.com>:

**Comment by jcchavezs** [comment]
@Landerstraeten is this still an issue?

**Comment by jcchavezs** [comment]
Thanks both @Landerstraeten and @UrGuardian4ngel!

**Pull request closed without merging**